utils/s3/aws_errors.cc

```diff
@@ -0,0 +1,149 @@
+    {"RequestThrottledException", aws_error(aws_error_type::THROTTLING, true)},
+    {"RequestThrottled", aws_error(aws_error_type::THROTTLING, true)},
+    {"ThrottlingException", aws_error(aws_error_type::THROTTLING, true)},
+    {"ThrottledException", aws_error(aws_error_type::THROTTLING, true)},
+    {"Throttling", aws_error(aws_error_type::THROTTLING, true)},
+    {"ValidationErrorException", aws_error(aws_error_type::VALIDATION, false)},
+    {"ValidationException", aws_error(aws_error_type::VALIDATION, false)},
+    {"ValidationError", aws_error(aws_error_type::VALIDATION, false)},
+    {"AccessDeniedException", aws_error(aws_error_type::ACCESS_DENIED, false)},
+    {"AccessDenied", aws_error(aws_error_type::ACCESS_DENIED, false)},
+    {"ResourceNotFoundException", aws_error(aws_error_type::RESOURCE_NOT_FOUND, false)},
+    {"ResourceNotFound", aws_error(aws_error_type::RESOURCE_NOT_FOUND, false)},
+    {"UnrecognizedClientException", aws_error(aws_error_type::UNRECOGNIZED_CLIENT, false)},
+    {"UnrecognizedClient", aws_error(aws_error_type::UNRECOGNIZED_CLIENT, false)},
+    {"SlowDownException", aws_error(aws_error_type::SLOW_DOWN, true)},
+    {"SlowDown", aws_error(aws_error_type::SLOW_DOWN, true)},
+    {"SignatureDoesNotMatchException", aws_error(aws_error_type::SIGNATURE_DOES_NOT_MATCH, false)},
+    {"SignatureDoesNotMatch", aws_error(aws_error_type::SIGNATURE_DOES_NOT_MATCH, false)},
+    {"InvalidAccessKeyIdException", aws_error(aws_error_type::INVALID_ACCESS_KEY_ID, false)},
+    {"InvalidAccessKeyId", aws_error(aws_error_type::INVALID_ACCESS_KEY_ID, false)},
+    {"RequestTimeTooSkewedException", aws_error(aws_error_type::REQUEST_TIME_TOO_SKEWED, true)},
+    {"RequestTimeTooSkewed", aws_error(aws_error_type::REQUEST_TIME_TOO_SKEWED, true)},
+    {"RequestTimeoutException", aws_error(aws_error_type::REQUEST_TIMEOUT, true)},
+    {"RequestTimeout", aws_error(aws_error_type::REQUEST_TIMEOUT, true)}};
+
+aws_error::aws_error(aws_error_type error_type, bool is_retryable) : _type(error_type), _is_retryable(is_retryable) {
+}
+
+aws_error::aws_error(aws_error_type error_type, std::string&& error_message, bool is_retryable)
+    : _type(error_type), _message(std::move(error_message)), _is_retryable(is_retryable) {
+}
+
+const std::string& aws_error::get_error_message() const {
+    return _message;
+}
+
+aws_error_type aws_error::get_error_type() const {
+    return _type;
+}
+
+bool aws_error::is_retryable() const {
+    return _is_retryable;
+}
+
+aws_error aws_error::parse(seastar::sstring&& body) {
+    aws_error ret_val;
+
+    if (body.empty())
+        return ret_val;
+
+    rapidxml::xml_document<> doc;
+    try {
+        doc.parse<0>(body.data());
+    } catch (const rapidxml::parse_error&) {
+        // Most likely not an XML which is possible, just return
+        return ret_val;
+    }
+
+    const auto* error_node = doc.first_node("Error");
+    if (!error_node) {
+        {
```

Comment:
why this extra indent?

Reply:
and extra scope :) no idea :) will remove, of course